server/src/main/java/org/apache/iotdb/db/metadata/MManager.java

```diff
@@ -603,36 +607,36 @@ public void setStorageGroup(String storageGroup) throws MetadataException {
   /**
    * Delete storage groups of given paths from MTree. Log format: "delete_storage_group,sg1,sg2,sg3"
    *
-   * @param storageGroups list of paths to be deleted. Format: root.node
+   * @param detachedStorageGroups list of paths to be deleted. Format: root.node
    */
-  public void deleteStorageGroups(List<String> storageGroups) throws MetadataException {
+  public void deleteStorageGroups(List<List<String>> detachedStorageGroups) throws MetadataException {
     lock.writeLock().lock();
     try {
-      for (String storageGroup : storageGroups) {
+      for (List<String> detachedStorageGroup : detachedStorageGroups) {
 
         // clear cached MNode
         mNodeCache.clear();
 
         // try to delete storage group
-        List<MeasurementMNode> leafMNodes = mtree.deleteStorageGroup(storageGroup);
+        List<MeasurementMNode> leafMNodes = mtree.deleteStorageGroup(detachedStorageGroup);
         for (MeasurementMNode leafMNode : leafMNodes) {
           removeFromTagInvertedIndex(leafMNode);
         }
 
         if (config.isEnableParameterAdapter()) {
           IoTDBConfigDynamicAdapter.getInstance().addOrDeleteStorageGroup(-1);
-          int size = seriesNumberInStorageGroups.get(storageGroup);
+          int size = seriesNumberInStorageGroups.get(MetaUtils.concatDetachedPathByDot(detachedStorageGroup));
           IoTDBConfigDynamicAdapter.getInstance().addOrDeleteTimeSeries(size * -1);
-          ActiveTimeSeriesCounter.getInstance().delete(storageGroup);
-          seriesNumberInStorageGroups.remove(storageGroup);
+          ActiveTimeSeriesCounter.getInstance().delete(MetaUtils.concatDetachedPathByDot(detachedStorageGroup));
+          seriesNumberInStorageGroups.remove(MetaUtils.concatDetachedPathByDot(detachedStorageGroup));
           if (size == maxSeriesNumberAmongStorageGroup) {
             maxSeriesNumberAmongStorageGroup =
                 seriesNumberInStorageGroups.values().stream().max(Integer::compareTo).orElse(0);
           }
         }
         // if success
         if (!isRecovering) {
-          logWriter.deleteStorageGroup(storageGroup);
+          logWriter.deleteStorageGroup(MetaUtils.concatDetachedPathByDot(detachedStorageGroup));
```

Comment:
Could we diff the two series  who's concat path is same, like following root."sg1.device1"."factory.one"."car.next"."measurement.high" and  root."sg1.device1"."factory.one.car.next"."measurement.high"

Reply:
I think It can.